homeassistant/components/totalconnect/__init__.py

```diff
@@ -1,58 +1,71 @@
 """The totalconnect component."""
+import asyncio
 import logging
 
 from total_connect_client import TotalConnectClient
-import voluptuous as vol
 
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
-from homeassistant.helpers import discovery
-import homeassistant.helpers.config_validation as cv
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-DOMAIN = "totalconnect"
+PLATFORMS = ["alarm_control_panel", "binary_sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up by configuration file."""
+    if DOMAIN not in config:
+        return True
 
-CONFIG_SCHEMA = vol.Schema(
```

Comment:
Why are we removing the config schema? We're still allowing import of the config to the config flow, so then we still need the config schema.

Reply:
Sorry.  I looked at multiple examples given (axis, hue, icloud) and and mine was similar.  I must not understand some inherited properties because CONFIG_SCHEMA is no longer used inside async_setup in any of them. I've re-added it.

Reply:
`CONFIG_SCHEMA` is used by the core component setup procedure that sets up all components. So it's happening behind the scenes.